### PR TITLE
VSC 1

### PR DESCRIPTION
**Pull request by zsd4yr**

### CODE REVIEW

File: source/guide.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+Using a VSC is important to all kinds of developers, from artists and programmers to composers and designers. Almost all the work we do as game makers is collaborative, and so having a way to easily trade files is important. But we could just use Google Drive or Dropbox for that. However, the additional functionality to look back on changes to files, who wrote (or broke) what, and the ability to retrieve a comprehensive history of files is what begins to set a VSC above traditional file methods. VSC's also, in many cases, allow multiple users to edit a file at the same time by merging the changes together later.
+
+### Important Terms and Commands
+
+For the rest of this guide, I'll be talking in terms of Github or Git, which is a single VSC among many. Many of the ideas of Git are shared among others, so this should simply this tutorial without causing too much confusion.
```
Comment (ferociouself) [inline source/guide.md]: simply -> simplify

File: source/guide.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+* Stash --> storing any edits to files or deletions of files (but not the creation of new files) on a stack (stacks are first on, last off. This means that if you stash A, B, C. They will return to you as C, B, A)
+* Stash Pop --> retrieving the last stash off the stack
+* Staged --> changes you wish to commit or stash
+* Unstaged --> changes you do not wish to commit or stash but do not with to discard
+* Discard --> permanently throwing away a local change before commiting
+* Revert --> undoing a push by returning the remote repository to an earlier state
+
+### Suggested Workflows
+
+It will be up to each project to decide which workflow is best for it. Almost always, an agreed-upon structure on each team is critical, but the exact nature of that structure is up to the needs and proficiencies of those involved. I have worked on projects where we pulled hundreds of branches and reviewed people's code before it could merge, and I've worked on R&D projects where the only think I ever did was commit straight into the master without any branching at all. Each project is different, but here are some heuristics that have served me well:
```
Comment (ferociouself) [inline source/guide.md]: think -> thing

File: source/guide.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+In this tactic, whenever anyone wants to do anything: add a feature, edit some code, upload art or music, delete a design document --they should pull a branch, commit changes into that branch, and then when they're ready to join everything back in, they should update that branch from the master branch and then submit a pull request (merge). Team leads, where appropriate, or the director, should then approve these. Usually these people can also commit straight into master, without cutting a branch, which will save the team time.
+
+##### Code Review
+
+Code Reviewing is the practice of having leads, usually engineers, look over peoples' code and suggesting fixes, changes, and bringing up possible bugs, edge cases, etc... that the original programmer may not have considered. Code Reviewers are usually skilled and experienced, but it may also be useful to have young programmers code review each others; even at a early stage, different programmers think differently and so this can be fruitful. The easiest way to code review is to look over commit history or check out pull requests (merges) from branches before approval.
+
+##### Pair Programming
```
Comment (ferociouself) [inline source/guide.md]: I kind of see how this relates to the heading directly above it, but I'm not sure this belongs in the Source Control article. The two above relate to VSC due to their integration with pull requests and branching, but pair programming seems to be a bit unrelated to the larger topic.

File: source/guide.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+* git remote -> Shows all the remote versions of your repository. Example: git remote origin
+* git log -> Shows a listing of commits on a branch including the corresponding details. Example: git log commit 84f241e8a0d768fb37ff7ad40e294b61a99a0abe Author: User <user@domain.com> Date: Mon May 3 09:24:05 2010 +0300 first commit
+* git show -> Shows information about a git object. Example: git show commit 84f241e8a0d768fb37ff7ad40e294b61a99a0abe Author: User <user@domain.com> Date: Mon May 3 09:24:05 2010 +0300 first commit diff --git a/README b/README new file mode 100644 index 0000000..e69de29
+* git ls-tree -> Shows a tree object, including the mode and the name of each item and the SHA-1 value of the blob or the tree that it points to. Example: git ls-tree master^{tree} 100644 blob e69de29bb2d1d6434b8b29ae775ad8c2e48c5391 README
+* git cat-file -> Used to view the type of an object through the SHA-1 value. Example: git cat-file -t e69de29bb2d1d6434b8b29ae775ad8c2e48c5391 blob
+* git grep -> Lets you search through your trees of content for words and phrases. Example: git grep "www.siteground.com" -- *.php
```
Comment (ferociouself) [inline source/guide.md]: You might need to escape that "*.php", it looks like it might cause some issues on the markup end.

**Review by ferociouself — CHANGES_REQUESTED** [review]
Some minor spelling corrections, and also a comment about the relevance of discussing Pair Programming in this article.

File: source/guide.md (hunk 1 above)
Comment (zsd4yr) [reply source/guide.md]: done

File: source/guide.md (hunk 2 above)
Comment (zsd4yr) [reply source/guide.md]: done

File: source/guide.md (hunk 3 above)
Comment (zsd4yr) [reply source/guide.md]: done

**Comment by zsd4yr** [comment]
@ferociouself you made an earlier comment about the php?

**Comment by ferociouself** [comment]
I don't remember anything about php, so I'm gonna go ahead and merge this in haha

**Pull request merged**